sdk/core/azure-core/azure/core/pipeline/transport/_base.py

```diff
@@ -409,12 +413,22 @@ def prepare_multipart_body(self):
         main_message.add_header("Content-Type", "multipart/mixed")
         if boundary:
             main_message.set_boundary(boundary)
-        for i, req in enumerate(requests):
+
+        for req in requests:
             part_message = Message()
-            part_message.add_header("Content-Type", "application/http")
-            part_message.add_header("Content-Transfer-Encoding", "binary")
-            part_message.add_header("Content-ID", str(i))
-            part_message.set_payload(req.serialize())
+            if req.multipart_mixed_info:
+                content_index = req.prepare_multipart_body(content_index=content_index)
+                part_message.add_header("Content-Type", req.headers['Content-Type'])
```

Comment:
Why we have add_header every time? Only the leaf one of "Content-Type" will take effect, right?

Reply:
Actually `Content-Type` is the one header that must be present for both parent changeset messages, and leaf sub-changeset messages. This is needed because the changeset message must declare the changeset boundary definition. You can see the results in the test here:
Changeset content-type:
https://github.com/Azure/azure-sdk-for-python/pull/10972/files#diff-102c265bbd4559f55d45afbd5dae4473R366

Leaf message content-type:
https://github.com/Azure/azure-sdk-for-python/pull/10972/files#diff-102c265bbd4559f55d45afbd5dae4473R369